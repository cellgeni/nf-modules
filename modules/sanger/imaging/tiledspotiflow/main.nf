process IMAGING_TILEDSPOTIFLOW {
    tag "${meta.id}"

    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? "quay.io/cellgeni/tiled_spotiflow:0.5.4-2"
        : "quay.io/cellgeni/tiled_spotiflow:0.5.4-2"}"

    input:
    tuple val(meta), val(x_min), val(y_min), val(x_max), val(y_max), path(image), val(ch_ind)

    output:
    tuple val(meta), val(ch_ind), path("${output_name}"), emit: peaks
    path "versions.yml", emit: versions

    script:
    def args = task.ext.args ?: ''
    output_name = "${meta.id}_ch_${ch_ind}_peaks_Y${y_min}_Y${y_max}_X${x_min}_X${x_max}.csv"
    """
    Spotiflow_call_peaks.py run \
        -image_path ${image} \
        -x_min ${x_min} \
        -y_min ${y_min} \
        -x_max ${x_max} \
        -y_max ${y_max} \
        -C ${ch_ind} \
        -output_name "${output_name}" \
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        IMAGING: \$(Spotiflow_call_peaks.py version)
    END_VERSIONS
    """

    stub:
    output_name = "${meta.id}_ch_${ch_ind}_peaks_Y${y_min}_Y${y_max}_X${x_min}_X${x_max}.csv"
    """
    touch "${output_name}"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        IMAGING: \$(Spotiflow_call_peaks.py version)
    END_VERSIONS
    """
}
