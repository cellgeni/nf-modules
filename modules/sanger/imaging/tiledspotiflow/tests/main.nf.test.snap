{
    "mouse_heart - tiff - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        0,
                        "mouse_heart_ch_0_peaks_Y0_Y200_X0_X200.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,5468a2cc58e86ec219e7b4f35377c575"
                ],
                "peaks": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        0,
                        "mouse_heart_ch_0_peaks_Y0_Y200_X0_X200.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5468a2cc58e86ec219e7b4f35377c575"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-03-14T11:16:20.49056051"
    },
    "mouse_heart - tiff": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        0,
                        "mouse_heart_ch_0_peaks_Y0_Y200_X0_X200.csv:md5,a1a930de917429f510577aa17970aaa6"
                    ]
                ],
                "1": [
                    "versions.yml:md5,5468a2cc58e86ec219e7b4f35377c575"
                ],
                "peaks": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        0,
                        "mouse_heart_ch_0_peaks_Y0_Y200_X0_X200.csv:md5,a1a930de917429f510577aa17970aaa6"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5468a2cc58e86ec219e7b4f35377c575"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-03-14T11:15:46.765297635"
    },
    "mouse_heart - tiff - 500": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        0,
                        "mouse_heart_ch_0_peaks_Y0_Y500_X0_X500.csv:md5,b751ba2081cc127d53c9457f5b15749d"
                    ]
                ],
                "1": [
                    "versions.yml:md5,5468a2cc58e86ec219e7b4f35377c575"
                ],
                "peaks": [
                    [
                        {
                            "id": "mouse_heart"
                        },
                        0,
                        "mouse_heart_ch_0_peaks_Y0_Y500_X0_X500.csv:md5,b751ba2081cc127d53c9457f5b15749d"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5468a2cc58e86ec219e7b4f35377c575"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-03-14T11:16:07.818311208"
    }
}